### PR TITLE
correction of login and return of json responses

### PR DESCRIPTION
**Pull request by joao-alegria**
Registration and Login supported by the platform.
Registration persists user in database and its interest areas.

### CODE REVIEW

**Comment by adebna** [comment]
There is a problem in the registration. You don't verify if the username is already taken.
That way it's possible to create 2 or more accounts with the same username and the login will fail to return only one row.

**Comment by aspedrosa** [comment]
Also take into account that an email is also unique. You should also check that before insert/register a user.

**Comment by joao-alegria** [comment]
Already corrected the vulnerabilities, as well as implemented the unit tests and the acceptance tests necessary.

File: rest_api/src/main/java/tqs/cloudit/services/AuthenticationService.java [hunk 1]
```diff
@@ -24,30 +23,49 @@
...
     private Logger l = LoggerFactory.getLogger(AuthenticationService.class);
     
     public ResponseEntity register(User user) {
+        if(!user.allDefined()){
+            JSONObject response = new JSONObject();
+            response.put("status", 1);
+            response.put("message", "Registration invalid. When registering you need to provide username, password, name, email, type of user and optionally interest areas.");
+            return ResponseEntity.badRequest().body(response);
+        }
+        if(userRepository.getUsernames().contains(user.getUsername())){
```
Comment (aspedrosa) [inline rest_api/src/main/java/tqs/cloudit/services/AuthenticationService.java]: I think you should leave this, checking if the username already exists in the user set, to SQL
```SQL
        SELECT count(*) FROM user WHERE username = ?1;
```
```suggestion
        if(userRepository.usernameExists() > 0){
```
Same applies to email

File: rest_api/src/main/java/tqs/cloudit/services/AuthenticationService.java [hunk 2]
```diff
@@ -24,30 +23,49 @@
...
-    }
-    
-    public ResponseEntity login(Credentials cred) {
-        if(this.bcpe.matches(cred.getPassword(),this.userRepository.getPass(cred.getUsername()))){
-            return ResponseEntity.ok("firmeza");
-        }
-        return ResponseEntity.ok("não firmeza");
+        
+        JSONObject response = new JSONObject();
+        response.put("status", 0);
```
Comment (aspedrosa) [inline rest_api/src/main/java/tqs/cloudit/services/AuthenticationService.java]: I think we should return the status of the response through HTTP status codes, what maybe needs to be discussed between the group.
```Java
        ResponseEntity(HttpStatus.Accepted, "Registered with success");
```

File: rest_api/src/main/resources/static/assets/js/login.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
+var base_api_url = "http://localhost:8080";
```
Comment (aspedrosa) [inline rest_api/src/main/resources/static/assets/js/login.js]:
```suggestion
var base_api_url = "http://192.168.160.63:8080";
```
Same to other js files with localhost

File: rest_api/nb-configuration.xml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (aspedrosa) [inline rest_api/nb-configuration.xml]: gitignore

File: rest_api/phantomjsdriver.log [hunk 5]
```diff
@@ -0,0 +1,2090 @@
+[INFO  - 2019-05-23T15:42:41.311Z] GhostDriver - Main - running on port 30998
```
Comment (aspedrosa) [inline rest_api/phantomjsdriver.log]: gitignore

**Comment by joao-alegria** [comment]
Those are some aspects that didn't occur to me, thanks. I'll add them as soon as possible.

**Pull request merged**